services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+         */
+        public Builder visibilityTimeout(Duration visibilityTimeout) {
+            this.visibilityTimeout = visibilityTimeout;
+            return this;
+        }
+
+        /**
+         * Define the amount of time, the receive call will block on the server waiting for messages to arrive if the
+         * queue is empty when the receive call is first made. This setting has no effect if long polling is disabled.
+         *
+         * @param longPollWaitTimeout The new longPollWaitTimeout value.
+         * @return This object for method chaining.
+         */
+        public Builder longPollWaitTimeout(Duration longPollWaitTimeout) {
+            this.longPollWaitTimeout = longPollWaitTimeout;
+            return this;
+        }
+
+        /**
+         * Define the minimum wait time for incoming receive message requests. Without a non-zero minimum wait time, threads can
+         * easily waste CPU time busy-waiting against empty local buffers. Avoid setting this to 0 unless you are confident
+         * threads will do useful work in-between each call to receive messages!
+         *
+         * @param minReceiveWaitTime The new minReceiveWaitTime value.
+         * @return This object for method chaining.
+         */
+        public Builder minReceiveWaitTime(Duration minReceiveWaitTime) {
+            this.minReceiveWaitTime = minReceiveWaitTime;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of concurrent receive message batches. The greater this number, the faster the queue will be
+         * pulling messages from the SQS servers (at the expense of consuming more threads).
+         *
+         * @param maxInflightReceiveBatches The new maxInflightReceiveBatches value.
+         * @return This object for method chaining.
+         */
+        public Builder maxInflightReceiveBatches(Integer maxInflightReceiveBatches) {
+            this.maxInflightReceiveBatches = maxInflightReceiveBatches;
+            return this;
+        }
+
+        /**
+         * Define the maximum number of done receive batches. If more than that number of completed receive batches are waiting in
+         * the buffer, the querying for new messages will stop. The larger this number, the more messages the buffer queue will
+         * pre-fetch and keep in the buffer on the client side, and the faster receive requests will be satisfied. The visibility
+         * timeout of a pre-fetched message starts at the point of pre-fetch, which means that while the message is in the local
+         * buffer it is unavailable for other clients to process, and when this client retrieves it, part of the visibility
+         * timeout may have already expired. The number of messages prefetched will not exceed maxBatchSize *
+         * maxDoneReceiveBatches.
+         *
+         * @param maxDoneReceiveBatches The new maxDoneReceiveBatches value.
+         * @return This object for method chaining.
+         */
```

Comment:
Some of these settings feel disconnected from what customers actually want. They probably don't care about max done received batches. They probably care more about how much memory is getting used.

Can we put it in those terms?